static/js/formatters-internal.js

```diff
@@ -281,134 +281,97 @@ export function joinList(list, separator) {
+    ? _replaceUrlHost(urlWithDesiredDims, 'dynl.mktgcdn.com')
+    : _replaceUrlHost(urlWithDesiredDims, 'dynm.mktgcdn.com');
 
   return Object.assign(
     {},
-    img,
+    image,
     {
-      url: result.replace('http://', 'https://')
+      url: dynamicUrl.replace('http://', 'https://')
     }
   );
 }
 
 /**
- * Gets the smallest thumbnail that is over the min width and min height.
- * If no thumbnails are over the given thresholds, will return the closest one.
- *
- * This method assumes all thumbnails have the same aspect ratio, and that
- * thumbnails are sorted in descending size.
- *
- * @param {Array<{{url: string, width: number, height: number}}>} thumbnails 
- * @param {number|undefined} minWidth 
- * @param {number|undefined} minHeight 
- * @returns {string}
+ * Splits a url into two parts at the specified index.
+ * 
+ * @param {string} url The url to be split
+ * @param {number} index The index at which to split the url
+ * @returns {Array<string>} The two parts of the url after splitting
  */
-function _getSmallestThumbnailOverThreshold(thumbnails, minWidth, minHeight) {
-  let index = thumbnails.length - 1;
-  while (index > 0) {
-    const thumb = thumbnails[index];
-    const widthOverThreshold = minWidth ? thumb.width >= minWidth : true;
-    const heightOverThreshold = minHeight ? thumb.height >= minHeight : true;
-    if (widthOverThreshold && heightOverThreshold) {
-      return thumb.url
-    }
-    index--;
-  }
-  return thumbnails[0].url;
+function _splitUrlOnIndex(url, index) {
+  return [url.slice(0, index), url.slice(index)];
 }
 
 /**
- * Gets the largest thumbnail that is under the max width and max height.
- * If no thumbnails are under the given thresholds, will return the closest one.
+ * Replaces the current host of a url with the specified host.
  * 
- * This method assumes all thumbnails have the same aspect ratio, and that
- * thumbnails are sorted in descending size.
- *
- * @param {Array<{{url: string, width: number, height: number}}>} thumbnails 
- * @param {number|undefined} maxWidth 
- * @param {number|undefined} maxHeight 
- * @returns {string}
+ * @param {string} url The url whose host is to be changed
+ * @param {string} host The new host to change to
+ * @returns {string} The url updated with the specified host
  */
-function _getLargestThumbnailUnderThreshold(thumbnails, maxWidth, maxHeight) {
-  let index = 0;
-  while (index < thumbnails.length) {
-    const thumb = thumbnails[index];
-    const widthOverThreshold = maxWidth ? thumb.width <= maxWidth : true;
-    const heightOverThreshold = maxHeight ? thumb.height <= maxHeight : true;
-    if (widthOverThreshold && heightOverThreshold) {
-      return thumb.url
-    }
-    index++;
-  }
-  return thumbnails[thumbnails.length - 1].url;
+function _replaceUrlHost(url, host) {
+  const splitUrl = url.split('://');
+  const urlAfterHost = splitUrl[1].slice(splitUrl[1].indexOf('/'));
+  return splitUrl[0] + '://'  + host + urlAfterHost;
```

Comment:
I wonder if there's a simple Regex.replace(...) that we can do here? If not that's fine. This is good too

Reply:
I think the regex would be something like `url.replace(/(\w+:\/\/)([^\/]+)(\/.*)/, '$1${host}$3')`. It seems harder to understand to me, but I can use that if it would be better

Reply:
I see, yeah that's alright then. Let's keep it as is